modules/structs/hook.go

```diff
@@ -40,17 +40,19 @@ type CreateHookOption struct {
 	// enum: gitea,gogs,slack,discord
 	Type string `json:"type" binding:"Required"`
 	// required: true
-	Config map[string]string `json:"config" binding:"Required"`
-	Events []string          `json:"events"`
+	Config       map[string]string `json:"config" binding:"Required"`
+	Events       []string          `json:"events"`
+	BranchFilter string            `json:"branch_filter" binding:"Regexp"`
```

Comment:
GlobPattern?

Reply:
Good catch, thanks

Reply:
Re-checked that validation works, just in case:

```
curl --header "Authorization: token XXX" "http://localhost:3000/api/v1/repos/WGH/test/hooks/3" -X PATCH -H "Content-Type: application/json" -d '{"Type": "slack", "Config": {"url": "http://foobar", "content_type": "json", "channel": "hi"}, "branch_filter": "[b-"}'
[{"fieldNames":["BranchFilter"],"classification":"GlobPattern","message":"unexpected end of input"}]
```